### PR TITLE
UIU-1600: Add interface checks for notes, feesfines and circulation

### PR DESCRIPTION
**Pull request by mkuklis**
This PR is trying to address issue described in: 

https://issues.folio.org/browse/UIU-1600

The user's details screen and edit form should now work without throwing bunch of errors when the `feefines` and `notes` interfaces are not present.

### CODE REVIEW

File: src/views/UserDetail/UserDetail.js [hunk 1]
```diff
@@ -444,19 +444,21 @@ class UserDetail extends React.Component {
...
-                  patronBlocks={patronBlocks}
-                  expanded={sections.patronBlocksSection}
-                  onToggle={this.handleSectionToggle}
-                  onClickViewPatronBlock={this.onClickViewPatronBlock}
-                  addRecord={this.state.addRecord}
-                  {...this.props}
-                />
-              </IfPermission>
+              <IfInterface perm="feefines">
+                <IfPermission perm="manualblocks.collection.get">
```
Comment (MikeTaylor) [inline src/views/UserDetail/UserDetail.js:448]: It's a real shame that everything inside the `<IfInterface>` has to be re-intented, so it's hard to see what has actually changed.

Oh well  ¯\\\_(ツ)_/¯

File: src/views/UserDetail/UserDetail.js [hunk 2]
```diff
@@ -444,19 +444,21 @@ class UserDetail extends React.Component {
...
-                  hasPatronBlocks={(hasPatronBlocks === 1 && totalPatronBlocks > 0)}
-                  patronBlocks={patronBlocks}
-                  expanded={sections.patronBlocksSection}
-                  onToggle={this.handleSectionToggle}
-                  onClickViewPatronBlock={this.onClickViewPatronBlock}
-                  addRecord={this.state.addRecord}
-                  {...this.props}
-                />
-              </IfPermission>
+              <IfInterface perm="feefines">
```
Comment (MikeTaylor) [inline src/views/UserDetail/UserDetail.js:447]: Surely `<IfInterface>` doesn't take a `perm` prop?!

Comment (mkuklis) [reply src/views/UserDetail/UserDetail.js]: good catch  @@MikeTaylor!

File: src/views/UserDetail/UserDetail.js (hunk 1 above)
Comment (mkuklis) [reply src/views/UserDetail/UserDetail.js:448]: @MikeTaylor I fixed the `perm` => `name` issue.

I'm not sure if we currently have anything in place to mark interfaces as optional do we? Does something like `okapiOptionalInterfaces` exist?

Comment (mkuklis) [reply src/views/UserDetail/UserDetail.js:448]: Ah we do have `optionalOkapiInterfaces` :)

File: package.json [hunk 3]
```diff
@@ -34,15 +34,18 @@
...
       "loan-storage": "4.0 5.0 6.0 7.0",
       "login": "6.0",
-      "feesfines": "15.0",
       "request-storage": "2.5 3.0",
       "users-bl": "5.0"
     },
+    "optionalOkapiInterfaces": {
+      "circulation": "9.0",
+      "feesfines": "15.0",
+      "notes": "1.0"
```
Comment (MikeTaylor) [inline package.json:47]: Oh, good catch on `notes`!

File: src/views/UserDetail/UserDetail.js (hunk 1 above)
Comment (MikeTaylor) [reply src/views/UserDetail/UserDetail.js:448]: Yep, I added `optionalOkapiInterfaces` myself! :-)

**Pull request merged**